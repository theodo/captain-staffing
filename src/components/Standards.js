import React, { Component } from 'react'
import PropTypes from 'prop-types'

export default class Standards extends Component {
    render() {
        const standards = this.props.weeks.map((week) => {
            const classNames = ['planning-cell', 'weekly-standard']

            if (this.props.weeklyTasksCount.hasOwnProperty(week.format('w')) && this.props.weeklyTasksCount[week.format('w')] > this.props.user.standards.projects) {
                classNames.push('error')
            }

            return (
                <div key={week.format('w')} className={classNames.join(' ')}></div>
            )
        })

        return (
            <div className="standards">{standards}</div>
        )
    }
}

Standards.propTypes = {
    user: PropTypes.shape({
        standards: PropTypes.shape({
            projects: PropTypes.number.isRequired
        }).isRequired
    }).isRequired,
    tasks: PropTypes.array.isRequired,
    weeks: PropTypes.array.isRequired,
    weeklyTasksCount: PropTypes.object.isRequired
}