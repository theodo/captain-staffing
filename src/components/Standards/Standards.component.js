import React, { Component } from 'react';
import PropTypes from 'prop-types';

export default class Standards extends Component {
  render() {
    const { weeklyTasksCount, user, weeks } = this.props;
    const { projects } = user.standards;

    const standards = weeks.map((week) => {
      const classNames = ['planning-cell', 'weekly-standard'];

      if (Object.prototype.hasOwnProperty.call(weeklyTasksCount, week.format('w')) && weeklyTasksCount[week.format('w')] > projects) {
        classNames.push('error');
      }

      return (
        <div key={week.format('w')} className={classNames.join(' ')} />
      );
    });

    return (
      <div className="standards">{standards}</div>
    );
  }
}

Standards.propTypes = {
  user: PropTypes.shape({
    standards: PropTypes.shape({
      projects: PropTypes.number.isRequired,
    }).isRequired,
  }).isRequired,
  tasks: PropTypes.array.isRequired,
  weeks: PropTypes.array.isRequired,
  weeklyTasksCount: PropTypes.object.isRequired,
};
