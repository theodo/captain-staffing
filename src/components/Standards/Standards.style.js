// @flow

import styled from 'styled-components';

export const StyledStandards = styled.div`
  position: absolute;
  left: 0;
  bottom: 0;
  display: flex;
  height: 5px;
`;

export const StyledWeeklyStandards = styled.div`
  display: flex;
  flex: 0 0 auto;
  width: 244px;
  z-index: 3;
  background-color: ${props => (props.isError ? 'red' : 'inherit')};
`;
