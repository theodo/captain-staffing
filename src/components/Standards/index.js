export { default } from './Standards.container';
