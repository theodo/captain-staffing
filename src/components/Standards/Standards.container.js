import { connect } from 'react-redux';
import Standards from './Standards.component';

const mapStateToProps = () => ({});

const mapDispatchToProps = () => ({});

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(Standards);

