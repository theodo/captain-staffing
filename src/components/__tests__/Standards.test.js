import React from 'react'
import { shallow, render } from 'enzyme'
import moment from 'moment'

import Standards from '../Standards'

describe('Standards test suites', () => {
    const user = {
        id: 1,
        username: "benjaming",
        standards: {
            projects: 1
        }
    }

    it('should render without throwing an error', () => {
        const standards = shallow(<Standards user={user} weeks={[]} weeklyTasksCount={{}} />)
        expect(standards.is('.standards')).toBe(true)
    })

    it('should render a grey line if there is no standard violation', () => {
        const weeks = [moment('14/08/2017', 'DD/MM/YYYY')]
        const weeklyTasksCount = {
            33: user.standards.projects
        }

        const standards = render(<Standards user={user} weeks={weeks} weeklyTasksCount={weeklyTasksCount} />)
        const weekStandard = standards.children().first()
        expect(weekStandard.html().match(/error/)).toBeFalsy()
    })
    
    it('should render a red line if there is a standard violation', () => {
        const weeks = [moment('14/08/2017', 'DD/MM/YYYY')]
        const weeklyTasksCount = {
            33: user.standards.projects + 1
        }

        const standards = render(<Standards user={user} weeks={weeks} weeklyTasksCount={weeklyTasksCount} />)
        const weekStandard = standards.children().first()
        expect(weekStandard.html().match(/error/)).toBeTruthy()
    })
})