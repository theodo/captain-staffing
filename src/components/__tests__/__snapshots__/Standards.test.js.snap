// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Standards test suites should render a grey line if there is no standard violation 1`] = `
.c0 {
  position: absolute;
  left: 0;
  bottom: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 5px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  width: 244px;
  z-index: 3;
  background-color: inherit;
}

<div
  className="c0"
>
  <div
    className="c1"
  />
</div>
`;

exports[`Standards test suites should render a red line if there is a standard violation 1`] = `
.c0 {
  position: absolute;
  left: 0;
  bottom: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 5px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  width: 244px;
  z-index: 3;
  background-color: red;
}

<div
  className="c0"
>
  <div
    className="c1"
  />
</div>
`;
